allo/backend/hls.py

```diff
@@ -278,8 +318,12 @@ def __init__(
                 self.host_code = ""
             with open(f"{project}/kernel.cpp", "w", encoding="utf-8") as outfile:
                 outfile.write(self.hls_code)
-            with open(f"{project}/host.cpp", "w", encoding="utf-8") as outfile:
-                outfile.write(self.host_code)
+            if platform.lower() != "ihls":
+                with open(f"{project}/host.cpp", "w", encoding="utf-8") as outfile:
+                    outfile.write(self.host_code)
+            else:
+                print("Platform is intel HLS; skipping generation of host.cpp.")
+
```

Comment:
The host code is also set as empty. No need to skip the generation. Otherwise, you'd better change the name of `kernel.cpp` as this one has already contained both host and device code.